src/main/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactory.java

```diff
@@ -210,8 +195,20 @@ private LinkedList<String> getReadStreamNames(ReadableConfig options) {
       ReadSession readSession =
           BigQueryReadSession.getReadsession(credentials, bqconfig, bigQueryReadClientFactory);
       List<ReadStream> readsessionList = readSession.getStreamsList();
-
-      // Iterate over read stream
+      Schema arrowReadSchema =
+          MessageSerializer.deserializeSchema(
+              new ReadChannel(
+                  new ByteArrayReadableSeekableByteChannel(
+                      readSession.getArrowSchema().getSerializedSchema().toByteArray())));
+      arrowReadSchema.toJson();
+      String fieldList = new String();
+      for (int i = 0; i < arrowReadSchema.getFields().size(); i++) {
+        fieldList = fieldList.concat(arrowReadSchema.getFields().get(i).getName() + ",");
+      }
+      bqconfig.setArrowSchemaFields(fieldList);
```

Comment:
FlinkBigQueryConfig should be immutable, please pass this in the from(...) method